libs/application-generic/src/usecases/delete-control-values/delete-control-values.usecase.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@nestjs/common';
+import { ControlValuesRepository } from '@novu/dal';
+import { ControlValuesLevelEnum } from '@novu/shared';
+import { DeleteControlValuesCommand } from './delete-control-values.command';
+import { Instrument, InstrumentUsecase } from '../../instrumentation';
+
+@Injectable()
+export class DeleteControlValuesUseCase {
+  constructor(private controlValuesRepository: ControlValuesRepository) {}
+
+  @InstrumentUsecase()
+  public async execute(command: DeleteControlValuesCommand): Promise<void> {
+    const existingControlValues = await this.controlValuesRepository.findOne({
```

Comment:
You can delete the values with a single query, instead of fetching and deleting. If the control is not found the delete query is just a Noop.